docs/usage.rst

```diff
@@ -4,32 +4,40 @@
 Usage
 =====
 
-Create a :class:`~parsel.selector.Selector` object for the HTML or XML text
-that you want to parse::
+Create a :class:`~parsel.selector.Selector` object for the input text that you want to parse and use.
 
-    >>> from parsel import Selector
-    >>> text = "<html><body><h1>Hello, Parsel!</h1></body></html>"
-    >>> selector = Selector(text=text)
-
-Then use `CSS`_ or `XPath`_ expressions to select elements::
+For HTML or XML, use `CSS`_ or `XPath`_ expressions to select elements::
 
-    >>> selector.css('h1')
+    >>> from parsel import Selector
+    >>> html_text = "<html><body><h1>Hello, Parsel!</h1></body></html>"
+    >>> html_selector = Selector(text=html_text)
+    >>> html_selector.css('h1')
     [<Selector xpath='descendant-or-self::h1' data='<h1>Hello, Parsel!</h1>'>]
-    >>> selector.xpath('//h1')  # the same, but now with XPath
+    >>> html_selector.xpath('//h1')  # the same, but now with XPath
     [<Selector xpath='//h1' data='<h1>Hello, Parsel!</h1>'>]
 
+For JSON format text, use `JMESPath`_ expressions to select fields inside the text::
+
+    >>> json_text = '{"title":"Hello, Parsel!"}'
+    >>> json_selector = Selector(text=json_text)
+    >>> json_selector.jmespath('title')
+    [<Selector xpath='title' data='Hello, Parsel!'>]
+
 And extract data from those elements::
 
-    >>> selector.css('h1::text').get()
+    >>> html_selector.css('h1::text').get()
     'Hello, Parsel!'
-    >>> selector.xpath('//h1/text()').getall()
+    >>> html_selector.xpath('//h1/text()').getall()
     ['Hello, Parsel!']
+    >>> json_selector.jmespath('title').get()
+    'Hello, Parsel!'
```

Comment:
I have just realized that the goal here was not to feature both CSS and XPath, but to feature both `get` and `getall`. So you could either leave it as it was, or maybe better, remove the XPath one and leave the JMESPath one using  `getall`.

Reply:
I think its good to display one simple example for each selector, as it completes the first level of the "tutorial" without leaving anything to imagination.
Maybe we could have 3 examples of get, one for each selector, and after them display do another line saying that getall can be used to get multiple data, and then display and example that brings multiple data, such as links?

Reply:
Maybe keep your change, and once we merge text support we can have one example here use it.

Reply:
I thought that it would be better to have all 3 examples retrieving the same thing, and briefly mention` getall()` as it feels that an explanation of it will suit it better than a 1 element list.

Reply:
I believe the point of this part is to show all ways to extract actual data. I am thinking that maybe we can leave this part exactly as it was, with only XPath and CSS, and we can later on add JMESPath if there is some other method to feature here.